src/Parser/ShadowPriest/Modules/Spells/Dispersion.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import Module from 'Parser/Core/Module';
+
+import SpellLink from 'common/SpellLink';
+
+import { formatPercentage } from 'common/format';
+import { calculateMaxCasts } from 'Parser/Core/getCastEfficiency';
+
+class Disperion extends Module {
+  _dispersions = {};
+  _previousDispersionCast = null;
+
+  on_initialized() {
+    this.active = true;
+  }
+
+  get uptime(){
+    return this.dispersionsAsArray.reduce((p, c) => p += c.end - c.start, 0);
```

Comment:
This looks to work the same as `this.owner.selectedCombatant.getBuffUptime(SPELLS.DISPERSION.id)`, would it be possible to replace this with that; removing the need to track the dispersion buffs manually?

Reply:
That particular function is indeed the same as getBuffUptime, I will get that changed. However, I use the dispersion start and end times in the Voidform graphs.